### PR TITLE
Add Support for Pip 23.1.1.

### PR DESCRIPTION
**Pull request by jsirois**
The release notes are here:
  https://pip.pypa.io/en/stable/news/#v23-1-1

### CODE REVIEW

**Comment by jsirois** [comment]
I merged #2132 in for its CI bitrot fixes and I won't submit this change until that one is merged.
The only commit to review here is 44b1610.

**Review by kaos — APPROVED** [review]

**Pull request merged**
